projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -161,12 +171,38 @@ public function update_submenus( $slug, $submenus_to_update ) {
 			return;
 		}
 
-		foreach ( $submenu[ $slug ] as $i => $submenu_item ) {
-			if ( array_key_exists( $submenu_item[2], $submenus_to_update ) ) {
-				$submenu_item[2] = $submenus_to_update[ $submenu_item[2] ];
-				// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-				$submenu[ $slug ][ $i ] = $submenu_item;
+		/**
+		 * Iterate over all submenu items and add the hide the submenus with CSS classes.
+		 * This is done separately of the second foreach because the position of the submenu might change.
+		 */
+		foreach ( $submenu[ $slug ] as $index => $item ) {
+			if ( ! array_key_exists( $item[2], $submenus_to_update ) ) {
+				continue;
+			}
+
+			$this->hide_submenu_element( $index, $slug, $item );
+		}
```

Comment:
Can we refactor this so we iterate the `$submenus_to_update` array rather than `$submenu[ $slug ]`? That way I think the code is a bit more readable.

```php
foreach ( array_keys( $submenus_to_update ) as $old_slug ) {
  $this->hide_submenu_page( $slug, $old_slug );
}
```

Reply:
Indeed it looks a bit more readable, but hide_submenu_page will iterate over the entire submenu item array on each method call while right now we are iterating just once; array_key_exists has a complexity of O(1) so it's pretty fast.

Imho, we should leave it as it is because it does not improve the readability that much. What do you think? :)

Reply:
That sounds good to me.

As a general note, I tend to favor readability over performance optimizations whenever is possible (i.e. if 99.99% of the time the submenu will not contain many items to make that optimization really noticeable), but it's just a personal preference since it'll probably make things easier for me if I have to go back to this code in 6 months (I wouldn't need to remember what's the difference between `hide_submenu_element ` and `hide_submenu_page`).

Reply:
Yeah, it's something that I prefer as well; a readable code has more added value than a code full of micro-optimizations that just makes harder to maintain.